server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -1119,6 +1103,42 @@ public IndexResult index(Index index) throws IOException {
         }
     }
 
+    @Override
+    public Engine.IndexResult addIndexOperationToTranslog(Index index) throws IOException {
+        IndexingStrategy plan = indexingStrategyForOperation(index);
```

Comment:
Why do you need to figure out the strategy for writes to translog?

Reply:
I was aiming to mirror the [construction of IndexResult after a Lucene indexing operation](https://github.com/opensearch-project/OpenSearch/blob/a2f00675aef61e431cffec5c4f36be97eb4c4302/server/src/main/java/org/opensearch/index/engine/InternalEngine.java#L1279) as closely as possible. This depends on the `versionForIndexing` and `currentNotFoundOrDeleted` values from the IndexingStrategy object, hence this call.